### PR TITLE
Show transfer view to user for approve with value

### PR DESCRIPTION
**Pull request by jpuri**
Improve approve transaction flow

https://app.zenhub.com/workspaces/confirmations-6245e6e2348677001213b8d2/issues/gh/metamask/metamask-planning/317

### CODE REVIEW

File: app/components/Nav/Main/RootRPCMethodsUI.js [hunk 1]
```diff
@@ -355,7 +355,11 @@ const RootRPCMethodsUI = (props) => {
           });
         }
 
-        if (data && data.substr(0, 10) === APPROVE_FUNCTION_SIGNATURE) {
+        if (
+          data &&
+          data.substr(0, 10) === APPROVE_FUNCTION_SIGNATURE &&
+          (!value || value === '0x0')
```
Comment (blackdevelopa) [inline app/components/Nav/Main/RootRPCMethodsUI.js:361]: This change looks good to me. Just a random thought, if we check for value === 0, shouldn't `isZero` work too?

Comment (jpuri) [reply app/components/Nav/Main/RootRPCMethodsUI.js:361]: here value is hex string so I am checking this

but totally valid point, can this be BigNumber? Are you suggesting to use `isZero` from some lib.

Comment (jpuri) [reply app/components/Nav/Main/RootRPCMethodsUI.js]: PR is updated

**Review by blackdevelopa — APPROVED** [review]
Lgtm. Audit is failing. You may need to rebase when the fix is updated to main

**Review by digiwand — APPROVED** [review]

**Comment by seaona** [comment]
Looks good! I land into the correct screen.

https://user-images.githubusercontent.com/54408225/220321115-9e700f9e-4e32-48e8-81c4-4dd99bebed93.mp4

**Comment by seaona** [comment]
@bschorchit pending to discuss RC target

**Pull request merged**